tests/testthat/test-validate_yaml.R

```diff
@@ -0,0 +1,240 @@
+
+test_that("empty staged dep yaml file throws error", {
+
+  text <- "---
+  "
+  expect_error(validate_staged_deps_yaml(yaml::read_yaml(text = text)))
+
+  text <- "
+  "
+  expect_error(validate_staged_deps_yaml(yaml::read_yaml(text = text)))
+})
+
+test_that("staged dep yaml file missing expected fields throws error", {
+
+  text <- "---
+  hello
+  "
+  expect_error(validate_staged_deps_yaml(yaml::read_yaml(text = text)))
+
+
+  text <- "---
+  downstream_repos:
+
+  upstream_repos:
+  "
+  expect_error(validate_staged_deps_yaml(yaml::read_yaml(text = text)))
+})
+
+test_that("staged dep yaml file without current_repo content throws error", {
+
+  text <- "---
+  downstream_repos:
+    - repo: test2/test
+      host: https://github.com
+  upstream_repos:
+
+  current_repo:
+
+  "
+  expect_error(validate_staged_deps_yaml(yaml::read_yaml(text = text)))
+})
+
+test_that("staged dep yaml file with current_repo as array throws error", {
+
+  text <- "---
+  upstream_repos:
+  downstream_repos:
+  current_repo:
+    - repo: test/test
+      host: https://github.com
+  "
+  expect_error(validate_staged_deps_yaml(yaml::read_yaml(text = text)))
+})
+
+test_that("staged dep yaml file cannot contain non-character values", {
+
+  text <- "---
+  upstream_repos:
+  downstream_repos:
+    - repo: test2/test
+      host: https://github.com
+  current_repo:
+    repo: true
+    host: https://github.com
+  "
+  expect_error(validate_staged_deps_yaml(yaml::read_yaml(text = text)))
+
+  text <- "---
+  upstream_repos:
+  downstream_repos:
+    - repo: test2/test
+      host: https://github.com
+  current_repo:
+    repo: 456
+    host: https://github.com
+  "
+  expect_error(validate_staged_deps_yaml(yaml::read_yaml(text = text)))
+
+  text <- "---
+  upstream_repos:
+  downstream_repos:
+    - repo: -5.645
+      host: https://github.com
+  current_repo:
+    repo: test/test
+    host: https://github.com
+  "
+  expect_error(validate_staged_deps_yaml(yaml::read_yaml(text = text)))
+
+  text <- "---
+  upstream_repos:
+  downstream_repos:
+    - repo: 546
+      host: https://github.com
+  current_repo:
+    repo: test/test
+    host: https://github.com
+  "
+  expect_error(validate_staged_deps_yaml(yaml::read_yaml(text = text)))
+})
+
+test_that("staged dep yaml file missing repo or host throws error", {
+
+  text <- "---
+  upstream_repos:
+  downstream_repos:
+    - repo: test2/test
+      host: https://github.com
+  current_repo:
+    repo: test/test
+    not_host: https://github.com
+  "
+  expect_error(validate_staged_deps_yaml(yaml::read_yaml(text = text)))
+
+  text <- "---
+  upstream_repos:
+  downstream_repos:
+    - repo: test2/test
+      host: https://github.com
+    - not_repo: test3/test
+      host: https://github.com
+  current_repo:
+    repo: test/test
+    host: https://github.com
+  "
+  expect_error(validate_staged_deps_yaml(yaml::read_yaml(text = text)))
+})
+
+test_that("staged dep yaml with array of current_repo is invalid", {
```

Comment:
this seems to be almost the same as the test starting on line 81.

Reply:
good spot - removing this